include/circle/reactive/signal.hpp

```diff
@@ -459,6 +477,43 @@ class scoped_connection
     connection c_;
 };
 
+class signal_blocker
+{
+public:
+    template <typename... Args>
+    signal_blocker(signal<Args...>& s) noexcept : connections_{s.connections_}
+    {
+        auto& connections = s.connections_;
+        const auto ids = connections->ids();
+        if (ids.empty())
+            CIRCLE_WARN("Creating signal_blocker without any connections");
+
+        for (auto id : ids)
+        {
+            if (!connections->block(id, true))
+                blocked_.insert(id);
+        }
+    }
+
+    signal_blocker(const signal_blocker&) = delete;
+    signal_blocker& operator=(const signal_blocker&) = delete;
+    signal_blocker(signal_blocker&&) = delete;
+    signal_blocker operator==(signal_blocker&&) = delete;
+
+    ~signal_blocker()
+    {
+        if (auto connections = connections_.lock())
+        {
+            for (auto id : blocked_)
+                connections->block(id, false);
+        }
+    }
+
+private:
+    std::weak_ptr<detail::connections_container_base> connections_;
+    detail::id_set blocked_;
```

Comment:
More straightforward solution than constructing `unordered_set` would be just introducing `bool connection_container::blocked` property. What do you think?

Reply:
You're absolutely right